mink/tasks/equality_constraint_task.py

```diff
@@ -0,0 +1,186 @@
+    return {
+        mujoco.mjtEq.mjEQ_CONNECT.value: 3,
+        mujoco.mjtEq.mjEQ_WELD.value: 6,
+        mujoco.mjtEq.mjEQ_JOINT.value: 1,
+        mujoco.mjtEq.mjEQ_TENDON.value: 1,
+    }[constraint]
+
+
+def _get_cost_dim(eq_types: np.ndarray) -> int:
+    """Get the total size of the cost vector for a set of equality constraints."""
+    dim: int = 0
+    for eq_type in eq_types:
+        dim += _get_constraint_dim(eq_type)
+    return dim
+
+
+def _get_equality_constraint_indices(
+    data: mujoco.MjData, eq_ids: np.ndarray
+) -> np.ndarray:
+    """Get indices of a given equality constraint if it is active."""
+    return np.isin(data.efc_id, eq_ids)
+
+
+class EqualityConstraintTask(Task):
+    """Regulate equality constraints in a model.
+
+    Equality constraints are useful, among other things, for modeling "loop joints"
+    such as four-bar linkages. In MuJoCo, there are several types of equality
+    constraints, including:
+
+    * ``mjEQ_CONNECT``: Connect two bodies at a point (ball joint).
+    * ``mjEQ_WELD``: Fix relative pose of two bodies.
+    * ``mjEQ_JOINT``: couple the values of two scalar joints
+    * ``mjEQ_TENDON``: couple the values of two tendons
+
+    This task can regulate all equality constraints in a model or a specific subset
+    identified by name or ID.
+
+    Attributes:
+        equalities: ID or name of the equality constraints to regulate. If not provided,
+            the task will regulate all equality constraints in the model.
+        cost: Cost vector for the equality constraint task. Either a scalar, in which
+            case the same cost is applied to all constraints, or a vector of shape
+            ``(neq,)``, where ``neq`` is the number of equality constraints in the
+            model.
+
+    Raises:
+        InvalidConstraint: If a specified equality constraint name or ID is not found.
+        TaskDefinitionError: If no equality constraints are found or if cost parameters
+            have invalid shape or values.
+
+    Example:
+
+    .. code-block:: python
+
+        # Regulate all equality constraints with the same cost.
+        eq_task = EqualityConstraintTask(model, cost=1.0)
+
+        # Regulate specific equality constraints with different costs.
+        eq_task = EqualityConstraintTask(
+            model,
+            cost=[1.0, 0.5],
+            equalities=["connect_right", "connect_left"]
+        )
+    """
+
+    def __init__(
+        self,
+        model: mujoco.MjModel,
+        cost: npt.ArrayLike,
+        equalities: Optional[Sequence[int | str]] = None,
+        gain: float = 1.0,
+        lm_damping: float = 0.0,
+    ):
+        self._mask: np.ndarray | None = None  # Active equality constraint mask.
+        self._neq_active: int | None = None  # Number of active equality constraints.
+
+        eq_ids: list[int] = []
+        if equalities is not None:
+            for eq_id_or_name in equalities:
```

Comment:
AFAICT if the user specifies the same constraint twice they'll get it duplicated. This will presumably work fine, maybe not considered an error? Up to you

Reply:
Fixed.